create config.properties, update UniversityDataManager constructor an…

AndreyLatishko opened the pull request:
…d UniversityMain main. have exception with url and not right driver

AndreyLatishko committed:
create config.properties, update UniversityDataManager constructor and UniversityMain main. have exception with url and not right driver

bfedoronchuk commented on src/main/java/task/UniversityDataManager.java:
```diff
@@ -15,13 +19,14 @@
-        props.setProperty("user", "root");
-        props.setProperty("serverTimezone", "UTC");
-        connection = DriverManager.getConnection(url, props);
+    public UniversityDataManager() throws SQLException, FileNotFoundException {
```
Pass props from main

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,15 +1,23 @@
-        } catch (Exception e) {
-            e.printStackTrace();
+    public static void main(String[] args) throws IOException {
+        try(InputStream inputStream = new FileInputStream("C:/Projects/testJDBC/src/config.properties")) {
```
Don't use nested try-catches. There is a redundant level of nesting.
Also, check if the file exists or not.

bfedoronchuk commented on src/main/java/task/UniversityDataManager.java:
```diff
@@ -15,13 +19,14 @@
-        props.setProperty("serverTimezone", "UTC");
-        connection = DriverManager.getConnection(url, props);
+    public UniversityDataManager() throws SQLException, FileNotFoundException {
+        try(InputStream inputStream = new FileInputStream("C:/Projects/testJDBC/src/config.properties")) {
```
You don't use InputStream below.

AndreyLatishko committed:
update UniversityMain main and UniversityDataManager constructer, have exception with url

bfedoronchuk commented on src/main/java/task/UniversityDataManager.java:
```diff
@@ -16,12 +17,11 @@
         Properties props = new Properties();
-        props.setProperty("password", "1234");
-        props.setProperty("user", "root");
+        props.setProperty("password", String.valueOf(UniversityMain.password));
```
Pass this values via constructor

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,11 +1,35 @@
+                password = Integer.parseInt(properties.getProperty("password"));
+                user = properties.getProperty("user");
+            }
+            UniversityDataManager universityDataManager = new UniversityDataManager();
```
try with resources

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,11 +1,35 @@
     public static void main(String[] args)  {
-        try(UniversityDataManager universityDataManager = new UniversityDataManager()){
+        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in))) {
+            System.out.println("Введите данные url, password, user");
```
Pass location of property file, not values directly

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,11 +1,35 @@
+                password = Integer.parseInt(bufferedReader.readLine());
+                user = bufferedReader.readLine();
+            } else {
+                System.out.println(1);
```
Redundant System.out.println

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,11 +1,35 @@
+import java.util.Properties;
 
 public class UniversityMain {
+    public static String url;
```
Don't store it here

AndreyLatishko committed:
create InfoProperties.java and update main but have new exception
java.lang.NumberFormatException: For input string: ""

AndreyLatishko committed:
create InfoProperties.java and update main but have new exception
java.lang.NumberFormatException: For input string: ""

bfedoronchuk commented on src/main/java/task/InfoProperties.java:
```diff
@@ -0,0 +1,30 @@
+package task;
+
+public class InfoProperties {
```
DbProperties

bfedoronchuk commented on src/main/java/task/UniversityDataManager.java:
```diff
@@ -15,11 +16,11 @@
-    public UniversityDataManager() throws SQLException {
-        String url = "jdbc:mysql://localhost:3306/test";
+
+    public UniversityDataManager(String url, int password, String user) throws SQLException {
```
Pass DbProperties

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,11 +1,38 @@
+        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in))) {
+            System.out.println("Введите путь к файлу:");
+            String fileName = bufferedReader.readLine();
+            if (fileName.equals("C://Projects//testJDBC//src//main//resources//config.properties")) {
```
https://docs.oracle.com/javase/7/docs/api/java/io/File.html#exists()

AndreyLatishko committed:
rename InfoProperties.java > DbProperties.java,
use File.exists() in main and delete input console
have exception with driver suitable
java.sql.SQLException: No suitable driver found for url="jdbc:mysql://localhost:3306/test"

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,11 +1,35 @@
-        try(UniversityDataManager universityDataManager = new UniversityDataManager()){
+    public static void main(String[] args) {
+        DbProperties dbProperties = new DbProperties();
+        String fileName = args[0];
```
Check if there are no args

AndreyLatishko committed:
refactor exception with SQL and query, update main, create Application.java but have nullpointerException with use Thread

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,13 +1,59 @@
+            e.getStackTrace();
+        }
+        DbProperties dbProperties = new DbProperties();
+        boolean returnValue = fileName.isEmpty();
```
naming

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,13 +1,59 @@
+    //C:\Projects\testJDBC\src\main\resources\config.properties
+    public static void main(String[] args) {
+        String fileName ="";
+        try {
```
Replace exception logic

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,13 +1,59 @@
+        DbProperties dbProperties = new DbProperties();
+        boolean returnValue = fileName.isEmpty();
+        if (!returnValue) {
+            File fileArgumentMain = new File(fileName);
```
naming

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,13 +1,59 @@
+            }
+        } else {
+            InputStream input = null;
+            try {
```
join both catches

bfedoronchuk commented on src/main/java/task/UniversityMain.java:
```diff
@@ -1,13 +1,59 @@
-    public static void main(String[] args)  {
-        try(UniversityDataManager universityDataManager = new UniversityDataManager()){
+    //C:\Projects\testJDBC\src\main\resources\config.properties
+    public static void main(String[] args) {
```
Everywhere you create File object - use exists() method

AndreyLatishko committed:
renaming and refactor logic fileName check

AndreyLatishko committed:
refactor main

AndreyLatishko committed:
add check null for properties info
